File: java/src/main/102BinaryTreeLevelOrderTraversal/102BinaryTreeLevelOrderTraversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> result = new ArrayList<List<Integer>>();
        if(root == null)
        {
            return result ;
        }
        Stack<TreeNode> stack = new Stack<TreeNode>();
        stack.push(root);
        while( ! stack.isEmpty())
        {
            List<TreeNode> tmp =  new ArrayList<TreeNode>();
            List<Integer> tmpNum  = new ArrayList<Integer>();
            while(! stack.isEmpty())
            {
                TreeNode i = stack.pop();
                tmp.add(i);
                tmpNum.add(i.val);
            }
            result.add(tmpNum);
            for(int i = tmp.size() -1; i >=0 ; i --)
            {
                if(tmp.get(i).right !=null)
                {stack.push(tmp.get(i).right);
                }
                if(tmp.get(i).left != null)
                {
                    stack.push(tmp.get(i).left);
                }
            }
        }
        return result ; 
        
    }
}

File: cpp/102BinaryTreeLevelOrderTraversal.cpp
/*
    file name: 102BinaryTreeLevelOrderTraversal.cpp
    Sitong Che
*/

/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>> ans;
        if(!root)
        {
            return ans;
        }

        queue<TreeNode*> q;
        q.push(root);
        while(!q.empty())
        {
            queue<TreeNode*> qq;
            vector<int> tmp;
            while(!q.empty())
            {
                TreeNode* cur = q.front();
                q.pop();
                tmp.push_back(cur->val);
                if(cur->left)
                {
                    qq.push(cur->left);
                }

                if(cur->right)
                {
                    qq.push(cur->right);
                }
            }

            ans.push_back(tmp);
            swap(q, qq);

        }
        
        return ans;
    }
};